### PR TITLE
Improve logger memory management

### PR DESCRIPTION
**Pull request by antrix1989**

### CODE REVIEW

File: IdentityCore/src/logger/MSIDLogger.m [hunk 1]
```diff
@@ -27,11 +27,13 @@
 #import "MSIDDeviceId.h"
 #import <pthread.h>
 
+static long s_maxQueueSize = 1000;
```
Comment (oldalton) [inline IdentityCore/src/logger/MSIDLogger.m]: Is 1000 a reasonable size? What do other libs use?

File: IdentityCore/src/logger/MSIDLogger.m [hunk 2]
```diff
@@ -27,11 +27,13 @@
...
+
 @interface MSIDLogger()
 
 @property (nonatomic) dispatch_queue_t loggerQueue;
+@property (nonatomic) dispatch_semaphore_t queueSemaphore;
 @property (nonatomic, copy) MSIDLogCallback callback;
-;
```
Comment (oldalton) [inline IdentityCore/src/logger/MSIDLogger.m]: Thanks for removing this, I saw it before but didn't want to submit a PR just to remove this :)

File: IdentityCore/src/logger/MSIDLogger.m (hunk 1 above)
Comment (antrix1989) [reply IdentityCore/src/logger/MSIDLogger.m]: cocoalumberjack has also 1k

Comment (oldalton) [reply IdentityCore/src/logger/MSIDLogger.m]: I was more wondering about default value. Let's keep it 1k for now and if necessary, we add configuration in future.

**Pull request merged**